mesh/mesh.cpp

```diff
@@ -1581,6 +1581,120 @@ void Mesh::SetAttributes()
    }
 }
 
+void Mesh::GetAttributeSetNames(std::set<std::string> &names) const
+{
+   names.clear();
+
+   std::map<std::string,Array<int> >::const_iterator it;
```

Comment:
minor:

Following [cpp core guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-for-range) I am wondering whether a range-for loop should be preferred here:

```
{
   names.clear();

   for(auto const &attr_set: attr_sets)
   {
      names.insert(attr_set.first);
   }
}   
```
? Same in some other places (GetBdrAttributeSetNames, ...)